core/src/main/java/org/elasticsearch/action/delete/TransportDeleteAction.java

```diff
@@ -130,18 +130,24 @@ protected Location onReplicaShard(DeleteRequest request, IndexShard indexShard)
     public static WriteResult<DeleteResponse> executeDeleteRequestOnPrimary(DeleteRequest request, IndexShard indexShard) {
         Engine.Delete delete = indexShard.prepareDeleteOnPrimary(request.type(), request.id(), request.version(), request.versionType());
         indexShard.delete(delete);
-        // update the request with the version so it will go to the replicas
-        request.versionType(delete.versionType().versionTypeForReplicationAndRecovery());
-        request.version(delete.version());
-
-        assert request.versionType().validateVersionForWrites(request.version());
-        DeleteResponse response = new DeleteResponse(indexShard.shardId(), request.type(), request.id(), delete.version(), delete.found());
-        return new WriteResult<>(response, delete.getTranslogLocation());
+        if (delete.getFailure() == null) {
+            // update the request with the version so it will go to the replicas
+            request.versionType(delete.versionType().versionTypeForReplicationAndRecovery());
+            request.version(delete.version());
+            assert request.versionType().validateVersionForWrites(request.version());
+            DeleteResponse response = new DeleteResponse(indexShard.shardId(), request.type(), request.id(), delete.version(), delete.found());
+            return new WriteResult<>(response, delete.getTranslogLocation());
+        } else {
+            return new WriteResult<>(delete.getFailure());
+        }
     }
 
     public static Engine.Delete executeDeleteRequestOnReplica(DeleteRequest request, IndexShard indexShard) {
         Engine.Delete delete = indexShard.prepareDeleteOnReplica(request.type(), request.id(), request.version(), request.versionType());
         indexShard.delete(delete);
+        if (delete.getFailure() != null) {
```

Comment:
hmm... I think we need to change replica to return a write result as well.. this is to ugly :( ( and I think we misunderstood each other on "not doing anything with the failures on replicas for now.. "